spec/requests/admin/lead_providers_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Admin::LeadProviders", type: :request do
+  let(:lead_provider) { FactoryBot.create(:lead_provider) }
+  let(:lead_provider_name) { Faker::Company.name }
+
+  describe "GET /admin/lead_providers" do
+    it "renders the correct template" do
+      # When
+      get "/admin/lead_providers"
+
+      # Then
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe "POST /admin/lead_providers" do
+    it "redirects to the list of providers on success" do
+      # When
+      post "/admin/lead_providers", params: { lead_provider: {
+        name: lead_provider_name,
+      } }
+
+      # Then
+      expect(response).to redirect_to("/admin/lead_providers")
+    end
+
+    it "creates a new lead provider" do
+      expect {
+        post "/admin/lead_providers", params: { lead_provider: {
+          name: lead_provider_name,
+        } }
+      }.to change { LeadProvider.count }.by(1)
```

Comment:
Is there a keyword `increase` that we could use?

Reply:
No, but this tests for increase. If the number decreased by one this test would fail. We would need `change { LeadProvider.count }.by(-1)